GS tests fails randomly.

**Describe the bug**
GS tests fails randomly.

**To Reproduce**
Steps to reproduce the behavior:
1. Install `piq`;
2. `pytest tests/test_gs.py`
3. See errors

**Expected behaviour**
Test should not fail.

**Additional context**
[link to crashed workflow](https://github.com/photosynthesis-team/piq/runs/875802482)
```
========================================================================================== test session starts ===========================================================================================
platform darwin -- Python 3.7.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /Users/prokopenko/Documents/repos/piq
collected 4 items                                                                                                                                                                                        

tests/test_gs.py ..F.                                                                                                                                                                              [100%]

================================================================================================ FAILURES ================================================================================================
___________________________________________________________________________________ test_similar_for_same_distribution ___________________________________________________________________________________

features_target_normal = tensor([[0.7327, 0.1664, 0.9767,  ..., 0.2672, 0.3089, 0.1489],
        [0.4704, 0.6700, 0.7041,  ..., 0.4117, 0.4750,...0.3366, 0.7831, 0.9925,  ..., 0.7527, 0.1652, 0.0172],
        [0.2119, 0.8782, 0.1128,  ..., 0.7998, 0.9881, 0.5772]])
features_prediction_normal = tensor([[0.0381, 0.0917, 0.8025,  ..., 0.0418, 0.4957, 0.7719],
        [0.7759, 0.9686, 0.8259,  ..., 0.9057, 0.1659,...0.4451, 0.6341, 0.6057,  ..., 0.1390, 0.0166, 0.9808],
        [0.8533, 0.9099, 0.4307,  ..., 0.3337, 0.4602, 0.2769]])

    @pytest.mark.skipif('gudhi' not in sys.modules, reason="Requires gudhi library")
    def test_similar_for_same_distribution(
            features_target_normal: torch.Tensor, features_prediction_normal: torch.Tensor) -> None:
        metric = GS(sample_size=1000, num_iters=100, i_max=1000, num_workers=4)
        diff = metric(features_prediction_normal, features_target_normal)
>       assert diff <= 2.0, \
            f'For same distributions GS should be small, got {diff}'
E       AssertionError: For same distributions GS should be small, got 3.704582050806065
E       assert tensor(3.7046, dtype=torch.float64) <= 2.0

tests/test_gs.py:53: AssertionError
======================================================================================== short test summary info =========================================================================================
FAILED tests/test_gs.py::test_similar_for_same_distribution - AssertionError: For same distributions GS should be small, got 3.704582050806065
================================================================================ 1 failed, 3 passed in 128.29s (0:02:08) =================================================================================
```

```
========================================================================================== test session starts ===========================================================================================
platform darwin -- Python 3.7.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /Users/prokopenko/Documents/repos/piq
collected 4 items                                                                                                                                                                                        

tests/test_gs.py .ssF                                                                                                                                                                              [100%]

================================================================================================ FAILURES ================================================================================================
_______________________________________________________________________________ test_differs_for_not_simular_distributions _______________________________________________________________________________

features_prediction_beta = tensor([[0.2755, 0.7426, 0.1578,  ..., 0.1302, 0.7292, 0.7280],
        [0.5066, 0.4837, 0.2609,  ..., 0.4531, 0.0891,...0.5726, 0.8885, 0.1629,  ..., 0.6592, 0.4184, 0.4696],
        [0.9013, 0.5129, 0.7186,  ..., 0.5779, 0.4343, 0.2548]])
features_target_normal = tensor([[0.7480, 0.0831, 0.1512,  ..., 0.3548, 0.4215, 0.2369],
        [0.2351, 0.8593, 0.1618,  ..., 0.9318, 0.1497,...0.0968, 0.6748, 0.4035,  ..., 0.9696, 0.4208, 0.3127],
        [0.0754, 0.4739, 0.4180,  ..., 0.6528, 0.7931, 0.3637]])

    @pytest.mark.skipif('gudhi' not in sys.modules, reason="Requires gudhi library")
    def test_differs_for_not_simular_distributions(
            features_prediction_beta: torch.Tensor, features_target_normal: torch.Tensor) -> None:
        metric = GS(sample_size=1000, num_iters=100, i_max=1000, num_workers=4)
        diff = metric(features_prediction_beta, features_target_normal)
>       assert diff >= 5.0, \
            f'For different distributions GS diff should be big, got {diff}'
E       AssertionError: For different distributions GS diff should be big, got 3.708508958419184
E       assert tensor(3.7085, dtype=torch.float64) >= 5.0

tests/test_gs.py:58: AssertionError
======================================================================================== short test summary info =========================================================================================
FAILED tests/test_gs.py::test_differs_for_not_simular_distributions - AssertionError: For different distributions GS diff should be big, got 3.708508958419184
=========================================================================== 1 failed, 1 passed, 2 skipped in 64.62s (0:01:04) ============================================================================
```

--- a/piq/gs.py
+++ b/piq/gs.py
@@ -7,10 +7,10 @@ https://arxiv.org/pdf/1802.02664.pdf
 from typing import Optional, Tuple
 from multiprocessing import Pool
 
-from scipy.spatial.distance import cdist
-
 import torch
+import gudhi
 import numpy as np
+from scipy.spatial.distance import cdist
 
 from piq.base import BaseFeatureMetric
 
@@ -97,18 +97,6 @@ def witness(
     Returns
         A list of persistence intervals and the maximal persistence value.
     """
-    # Install gudhi only if needed
-    try:
-        import gudhi
-    except ImportError as e:
-        import six
-        error = e.__class__(
-            "You are likely missing your GUDHI installation, "
-            "you should visit http://gudhi.gforge.inria.fr/python/latest/installation.html "
-            "for further instructions.\nIf you use conda, you can use\nconda install -c conda-forge gudhi"
-        )
-        six.raise_from(error, e)
-
     N = features.shape[0]
     if gamma is None:
         gamma = 1.0 / 128 * N / 5000

